presto-main/src/main/java/io/prestosql/server/ExecutingStatementResourceConfig.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.server;
+
+import io.airlift.configuration.Config;
+import io.airlift.configuration.ConfigDescription;
+import io.airlift.units.DataSize;
+
+import javax.annotation.Nullable;
+
+import java.util.Optional;
+
+import static io.airlift.units.DataSize.Unit.MEGABYTE;
+
+public class ExecutingStatementResourceConfig
+{
+    private Optional<DataSize> serverTargetResultSize;
```

Comment:
Thanks for raising this PR. Instead of having it as `Optional` what if we could set it as the default value (`1MB`) in this case 

```
        precedence:
        1. passed to session
        2. server side config
        3. default
```
we can merge step 2 and 3. If we set a value then we would use it else we might use the default value.

Reply:
If we are planning to use `Optional` we can initialise it as `Optional.ofEmpty` instead of null

Reply:
updated